sympy/simplify/simplify.py

```diff
@@ -1241,7 +1241,7 @@ def cancel_kronecker_mul(m):
             args1 = delta1.args
             args2 = delta2.args
             if args_cancel(args1, args2):
-                return 0*m
```

Comment:
this one makes me a little nervous since `0*obj` might not return 0: it might return a type of zero as represented by that object.
```python
>>> Matrix([[1,2]])*0
Matrix([[0, 0]])
```

Reply:
Surely this function only operates on Expr objects

Reply:
```python
>>> var('i', integer=1)
i
>>> simplify(eye(1)*KroneckerDelta(0, i) * KroneckerDelta(1, i))
Matrix([[0]])
```

Reply:
Yeah, I removed it to see if any tests failed. Which didn't happen. But we should probably add a test like above. I'll do that.